### ISSUE TITLE
make_trackhubs error with BAM table key

### ISSUE THREAD

**Issue by nsheff** [pepkit/peppy#14]
```
Traceback (most recent call last):
  File "/home/nsheffield/code//looper/scripts/make_trackhubs.py", line 501, in <module>
    html_out_tab += '<td><a href="{}"><b>{}</b></a></td>\n'.format(value['BAM']['link'],value['BAM']['label'])
KeyError: 'BAM'
make: *** [trackhub] Error 1
```

```

This appears to be expecting a specific, undocumented format in the config file? 
        html_out_tab += '<td><a href="{}"><b>{}</b></a></td>\n'.format(value['BAM']['link'],value['BAM']['label'])
        html_out_tab += '<td><a href="{}"><b>{}</b></a></td>\n'.format(value['BAI']['link'],value['BAI']['label'])
        html_out_tab += '<td><a href="{}"><b>{}</b></a></td>\n'.format(value['BB']['link'],value['BB']['label'])
        html_out_tab += '<td><a href="{}"><b>{}</b></a></td>\n'.format(value['BW']['link'],value['BW']['label'])
        html_out_tab += '<td><a href="{}"><b>{}</b></a></td>\n'.format(value['BED']['link'],value['BED']['label'])
```

@cdietzgit ? Am I doing something wrong?

**Comment by cdietzgit** [comment]
Most likely you didn't do anything wrong. I guess it didn't find the path to the BAM file and since the fields for the HTML report are hard-coded and non-resilient to one field missing it crashes. The trackhub file should be there though.

Hum, the other day I had to produce a report for an old WGBS project using the script and I had to hack my own script to get it work (I forked it and I hard-coded what I needed for that I didn't have time to do more). Been thinking of some way that script could produce trackhubs AND html report page for our main pipelines (RRBS, WGBS, rna-stuff, etc.), could be useful. But I haven't started implementing it yet...
